src/vivarium_gates_child_iv_iron/components/fertility.py

```diff
@@ -34,28 +31,28 @@ def __repr__(self):
     def name(self):
         return "line_list_fertility"
 
-    def setup(self, builder):
+    # noinspection PyAttributeOutsideInit
+    def setup(self, builder: Builder):
         self.clock = builder.time.clock()
-        self.randomness = builder.randomness
         self.simulant_creator = builder.population.get_simulant_creator()
 
         # Requirements for input data
-        self.fertility_data_directory = builder.configuration.input_data.fertility_input_data_path
-        self.draw = builder.configuration.input_data.input_draw_number
-        self.seed = builder.configuration.randomness.random_seed
-        self.scenario = builder.configuration.intervention.scenario
-        self.birth_records = self._get_birth_records()
+        self.birth_records = self._get_birth_records(builder)
 
         builder.event.register_listener("time_step", self.on_time_step)
 
-    def _get_birth_records(self) -> pd.DataFrame:
+    @staticmethod
+    def _get_birth_records(builder: Builder) -> pd.DataFrame:
         """
         Method to load existing fertility data to use as birth records.
         """
-        fertility_data_dir = self.fertility_data_directory
-        file_path = fertility_data_dir + f'scenario_{self.scenario}_draw_{self.draw}_seed_{self.seed}.hdf'
-        birth_records = pd.read_hdf(file_path)
+        data_directory = Path(builder.configuration.input_data.fertility_input_data_path)
```

Comment:
The only real change here is using `Path` to define the data directory rather than using `str` and using the path combiner `/` below in line 54

Reply:
Did you move draw and seed declaration to local variables for the static method because this is the only instance we need them in this module?